pkg/apis/core/v1beta1/validation/validation_test.go

```diff
@@ -0,0 +1,290 @@
+		propagationController v1beta1.ControllerStatus
+		statusController      *v1beta1.ControllerStatus
+		expectedErr           bool
+		expectedErrMsg        string
+	}{
+		{
+			name:                  "valid status",
+			observedGeneration:    1,
+			propagationController: running,
+			statusController:      &notRunning,
+			expectedErr:           false,
+		},
+		{
+			name:               "PropagationController required",
+			observedGeneration: 1,
+			statusController:   &running,
+			expectedErr:        true,
+			expectedErrMsg:     "status.propagationController: Required value",
+		},
+		{
+			name:                  "negative ObservedGeneration",
+			observedGeneration:    -1,
+			propagationController: running,
+			statusController:      &notRunning,
+			expectedErr:           true,
+			expectedErrMsg:        "must be greater than or equal to 0",
+		},
+		{
+			name:                  "invalid PropagationController value",
+			observedGeneration:    1,
+			propagationController: invalidControllerStatus,
+			statusController:      &running,
+			expectedErr:           true,
+			expectedErrMsg:        "status.propagationController: Unsupported value",
+		},
+		{
+			name:                  "invalid StatusController value",
+			observedGeneration:    1,
+			propagationController: running,
+			statusController:      &invalidControllerStatus,
+			expectedErr:           true,
+			expectedErrMsg:        "status.statusController: Unsupported value",
+		},
+	}
+
+	for _, test := range testCases {
+		status := &v1beta1.FederatedTypeConfigStatus{
+			ObservedGeneration:    test.observedGeneration,
+			PropagationController: test.propagationController,
+			StatusController:      test.statusController,
+		}
+
+		errs := ValidateFederatedTypeConfigStatus(status, field.NewPath("status"))
+		hasErr := len(errs) > 0
+		if hasErr && hasErr != test.expectedErr {
+			t.Errorf("[%s] expected failure", test.expectedErrMsg)
+		} else if hasErr && !strings.Contains(errs[0].Error(), test.expectedErrMsg) {
+			t.Errorf("unexpected error: %q, expected: %q", errs[0].Error(), test.expectedErrMsg)
+		}
+	}
+
+}
+
+func successCases() []*v1beta1.FederatedTypeConfig {
+	return []*v1beta1.FederatedTypeConfig{
+		federatedTypeConfig(apiResourceWithEmptyGroup()),
+		federatedTypeConfig(apiResourceWithNonEmptyGroup()),
+	}
+}
+
+func apiResourceWithEmptyGroup() *metav1.APIResource {
+	return &metav1.APIResource{
+		Group:      "",
+		Version:    "v1",
+		Kind:       "Service",
+		Name:       "services",
+		Namespaced: true,
+	}
+}
+
+func apiResourceWithNonEmptyGroup() *metav1.APIResource {
+	return &metav1.APIResource{
+		Group:      "apps",
+		Version:    "v1",
+		Kind:       "Deployment",
+		Name:       "deployments",
+		Namespaced: true,
+	}
+}
+
+func validFederatedTypeConfig() *v1beta1.FederatedTypeConfig {
+	return federatedTypeConfig(apiResourceWithNonEmptyGroup())
+}
+
+func federatedTypeConfig(apiResource *metav1.APIResource) *v1beta1.FederatedTypeConfig {
```

Comment:
Yes, as it's where it originally comes from. :smile: Only thing is I didn't want to add more code changes by reworking the `kubefedctl` one as it incorporates defaulting and this one also includes `Status`. Once defaulting is added it can be revisited.